### PR TITLE
Fix pointer arithmetic error by shift operation

### LINKED ISSUE #21: incorrect difference of address
In test file `tests/ptr.c`, the execution result of following code is incorrect:

```
s = (int*)0xbebebeb0;
e = (int*)0xbebebeb4;
v = e - s;
if (v == 1)
  printf("passed\n");
else
printf("failed, e - s = %x\n", v);
```

Which print `failed, e - s = 0`

### PR DESCRIPTION
**Pull request by JohnnyLChang**
Fixes #21"

### CODE REVIEW

**Comment by jserv** [comment]
Rework commit message: `Fixes #21" in order to close issue #21.` is not effective. Instead, simply say "Fixes #21".

**Comment by jserv** [comment]
Please read [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/) carefully and rewrite the commit message.

**Comment by JohnnyLChang** [comment]
I appreciate your assistance.

**Comment by jserv** [comment]
Change the Git commit message again. You didn't fully apply the advised changes.

**Comment by JohnnyLChang** [comment]
got it, thanks

File: amacc.c [hunk 1]
```diff
@@ -408,7 +408,10 @@ void expr(int lev)
             sz = t >= PTR2 ? sizeof(int) :
                              t >= PTR ? tsize[t - PTR] : 1;
             if (t == ty && sz > 1) {
-                *++e = SUB; *++e = PSH; *++e = IMM; *++e = sz; *++e = SUB;
+                switch(sz){
```
Comment (jserv) [inline amacc.c]: Indent `switch` statement.

Comment (jserv) [inline amacc.c]: I expect `switch (sz) {" for consistent style.

File: amacc.c [hunk 2]
```diff
@@ -408,7 +408,10 @@ void expr(int lev)
             sz = t >= PTR2 ? sizeof(int) :
                              t >= PTR ? tsize[t - PTR] : 1;
             if (t == ty && sz > 1) {
-                *++e = SUB; *++e = PSH; *++e = IMM; *++e = sz; *++e = SUB;
+                switch(sz) {
```
Comment (jserv) [inline amacc.c]: There should be a space before the open parenthesis. That is, `switch (sz) {` rather than `switch(sz) {`.

**Comment by jserv** [comment]
@JohnnyLChang , Thanks for your contribution. Please check the commit message for writing better one.

**Comment by JohnnyLChang** [comment]
I really appreciate for your information and I will review it. thanks.

2017-06-09 16:43 GMT+08:00 Jim Huang <notifications@github.com>:

**Pull request merged**